packages/cf-builder-card/src/cardReducer.js

```diff
@@ -1,10 +1,12 @@
-const ActionTypes = require('./CardActionTypes');
+// @flow
 
-const initialState = {};
+import type {CardAction, CardState} from './CardTypes';
```

Comment:
The ES2015/6 module semantics are clearly defined in chapter 8, 9, and 15 in the ECMAScript 2015/6 spec.

AFAIK from November TC39 minutes, they mainly talked about [node module interops with ESM](https://github.com/nodejs/node-eps/blob/master/002-es6-modules.md), dynamic imports and whether `export * from` should export a default as well.

For node/ESM interops, the things at stake are 1) how to parse the source unambiguously to distinguish whether it's a script or a module, 2) deal with unbound names in circular import. Both of which are stage 1. I don't want to go into details but AFAIK, 1) has [an answer](https://github.com/nodejs/node-eps/blob/5dae5a537c2d56fbaf23aaf2ae9da15e74474021/002-es6-modules.md#51-determining-if-source-is-an-es-module) in Node, they are just trying to find out if it would be useful for the Web as well. 2) also has 3 options, no breaking change was proposed.

Webpack (as of rc3, rc4 broke this) seems to follow the Node EPS closely with some necessary small differences. We would love to enable downstream users to tree-shake cf-ui. We can only do that with import/export so Webpack can compile into a form that it can tree-shake. Not that we have much to tree-shake, but it'd be nice.

There's no plan to change module semantics for dynamic imports, Chrome is implementing it now. We are currently using Babel's dynamic import syntax plugin to pass `import()` thru to Webpack.

The consensus for exporting the default in `export * from` is there isn't consensus, so we are still in the steady state that's ES2015/6 module semantics, and the `export default from` and `export * as ns from` stage 1 proposals. I assume you are talking about this when you said "break things", which it might if it were adopted.

If you are talking about a Babel module transform bug, I can tell you for sure we are not going to transform ES6 module with Babel. `{module: false}` will most certainly be in `.babelrc`.

If you are talking about ES6 modules not implemented in browsers, that's because they are all waiting for the dynamic imports/loader spec, in which the unambiguous grammar issue kinda a big deal.